util/conditions/v1beta2/getter.go

```diff
@@ -166,19 +166,26 @@ func convertFromUnstructuredConditions(conditions []interface{}) ([]metav1.Condi
 // also, only a few, minimal rules are enforced, just enough to allow surfacing a condition from a providers object with Mirror.
 func validateAndFixConvertedCondition(c *metav1.Condition) error {
 	if c.Type == "" {
-		return errors.New("condition type must be set")
+		return errors.New("type must be set for all conditions")
 	}
 	if c.Status == "" {
-		return errors.New("condition status must be set")
+		return errors.Errorf("status must be set for the %s condition", c.Type)
 	}
+	switch c.Status {
+	case metav1.ConditionFalse, metav1.ConditionTrue, metav1.ConditionUnknown:
+		break
+	default:
+		return errors.Errorf("status for the %s condition must be one of %s, %s, %s", c.Type, metav1.ConditionTrue, metav1.ConditionFalse, metav1.ConditionUnknown)
+	}
+
 	if c.Reason == "" {
 		switch c.Status {
 		case metav1.ConditionFalse: // When using old Cluster API condition utils, for conditions with Status false, Reason can be empty only when a condition has negative polarity (means "good")
 			c.Reason = NoReasonReported
 		case metav1.ConditionTrue: // When using old Cluster API condition utils, for conditions with Status true, Reason can be empty only when a condition has positive polarity (means "good").
 			c.Reason = NoReasonReported
 		case metav1.ConditionUnknown:
-			return errors.New("condition reason must be set when a condition is unknown")
+			return errors.Errorf("reason must be set for the %s condition, status unknown", c.Type)
```

Comment:
What does the status have to do with that reason is missing?

Reply:
Good catch. I think it was a left over from an older version where we were using different reason based on status, but now this entire switch does not make sense anymore.
Dropped